File: 14_Web/08.05.2024/youtube_spring/src/model/UserDetails.ts
export class UserDetails{
    public id:number;
    public email:string;
    public password:string;
    public userType:string;

    constructor(id:number,email:string,password:string,userType:string){
        this.id=id;
        this.email=email;
        this.password=password;
        this.userType=userType;
    }
}

File: 12_C2C/songs/src/main/java/com/mindali/songs/beans/UserDetails.java
package com.mindali.songs.beans;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
public class UserDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String email;
    private String password;
    private UserType userType;

    public UserDetails(int id, String email, String password, UserType userType) {
        this.id = id;
        this.email = email;
        this.password = password;
        this.userType = userType;
    }

    public UserDetails(String email, String password, UserType userType){
        this.email=email;
        this.password=password;
        this.userType=userType;
    }
}
